docs/book/v2/migration.md

```diff
@@ -0,0 +1,12 @@
+# Migration from version 1
+
+This documents details changes made from version 1 of this package or from its implementation in `laminas-mvc`.
+
+## PSR-15 Middleware
+
+Only PSR-15 style middleware can be dispatched anymore, so all middleware has to implement
+`\Psr\Http\Server\MiddlewareInterface` or `\Psr\Http\Server\RequestHandlerInterface` 
+
+## Removed Route match params from request
+Previously, all matched route params were added to the `ServerRequestInterface` as attributes. This is no longer the case.
+If you need route params, you can extract them from the `\Laminas\Router\RouteMatch` request attribute which is still available.
```

Comment:
I think something like this would emphasise the change better
```suggestion
Previously, for the matched route both `Laminas\Router\RouteMatch` object as well as individual matched params were added to PSR-7 request as attributes. This is no longer the case and only the RouteMatch object can be accessed as PSR-7 request attribute: `$request->getAttribute(\Laminas\Router\RouteMatch::class);`
```

```